idunn/geocoder/models.py

```diff
@@ -0,0 +1,177 @@
+"""
+Implement GeocodeJson specification as defined here:
+ - https://github.com/geocoders/geocodejson-spec/tree/master/draft
+ - https://github.com/CanalTP/mimirsbrunn/blob/master/libs/bragi/src/model.rs
+"""
+import json
+
+from enum import Enum
+from typing import Any, List, Optional, Tuple
+
+from pydantic import BaseModel, PositiveInt, confloat
+
+
+Lon = confloat(ge=-180, le=180)
+Lat = confloat(ge=-90, le=90)
+
+Rect = Tuple[Lon, Lat, Lon, Lat]
+
+
+class Coord(BaseModel):
+    lon: Lon
+    lat: Lat
+
+
+class PoiType(BaseModel):
+    id: str
+    name: str
+
+
+class CommercialMode(BaseModel):
+    id: str
+    name: str
+
+
+class PhysicalMode(BaseModel):
+    id: str
+    name: str
+
+
+class Network(BaseModel):
+    id: str
+    name: str
+
+
+class Code(BaseModel):
+    name: str
+    value: str
+
+
+class ZoneType(str, Enum):
+    Suburb = 'suburb'
+    CityDistrict = 'city_district'
+    City = 'city'
+    StateDistrict = 'state_district'
+    State = 'state'
+    CountryRegion = 'country_region'
+    Country = 'country'
+    NonAdministrative = 'non_administrative'
+
+
+class AssociatedAdmin(BaseModel):
+    id: str
+    insee: str
+    level: PositiveInt
+    label: str
+    name: str
+    zip_codes: List[str]
+    coord: Coord
+    bbox: Optional[Rect]
+    zone_type: Optional[ZoneType]
+    parent_id: Optional[str]
+    codes: List[Code]
+
+
+class Line(BaseModel):
+    id: str
+    name: str
+    code: Optional[str]
+    color: Optional[str]
+    text_color: Optional[str]
+    commercial_mode: Optional[CommercialMode]
+    network: Optional[Network]
+    physical_modes: Optional[PhysicalMode]
+    sort_order: Optional[PositiveInt]
+
+
+class FeedPublished(BaseModel):
+    id: str
+    licence: str
+    name: str
+    url: str
+
+
+class GeocodingProperty(BaseModel):
+    key: str
+    value: str
+
+
+class Comment(BaseModel):
+    name: str
+
+
+class GeocodingResponse(BaseModel):
+    type: str
+    label: Optional[str]
+    name: Optional[str]
+    housenumber: Optional[str]
+    street: Optional[str]
+    locality: Optional[str]
+    postcode: Optional[str]
+    city: Optional[str]
+    #  accuracy: Optional[PositiveInt]
```

Comment:
Please remove those fields if they're not used.

Reply:
The thing is that they are in the "specification" (https://github.com/geocoders/geocodejson-spec), but they are commented out in Bragi: https://github.com/CanalTP/mimirsbrunn/blob/master/libs/bragi/src/model.rs#L194-L199.

I should have at least added a comment about that but couldn't it be relevant to keep them here in order to "explain what is missing from the standard" while avoiding using them when Bragi doesn't ?

Reply:
Hum... Then add a comment over the commented fields so people don't wonder why they are still here.